modules/nw-osp-pod-adding-connections-ipv6.adoc

```diff
@@ -0,0 +1,52 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/network-configuration.adoc
+
+:_content-type: PROCEDURE
+[id="nw-osp-pod-adding-connections-ipv6_{context}"]
+= Adding IPv6 connectivity to pods on {rh-openstack}
+
+After you enable IPv6 connectivity in pods, add connectivity to them by using a Container Network Interface (CNI) configuration.
+
+.Procedure
+
+. Create a file named `network.yaml`.
+
+. Specify your CNI configuration. For example, the following CNI configuration uses a SLAAC address mode with MACVLAN:
++
+[source,yaml]
+----
+spec:
+  additionalNetworks:
+  - name: ipv6
+    namespace: ipv6 <1>
+    rawCNIConfig: '{ "cniVersion": "0.3.1", "name": "ipv6", "type": "macvlan", "master": "ens4"}' <2>
+    type: Raw
+----
+<1> Be sure to create pods in this same namespace.
+<2> The interface in the network attachment `"master"` field can differ from `"ens4"` when more networks are configured or when a different kernel driver is used.
++
+[NOTE]
+====
+If you are using stateful address mode, include the IP Address Management (IPAM) in the CNI configuration.
+
+DHCPv6 is not supported by Multus.
+====
+
+. On a command line, enter:
++
+[source,terminal]
+----
+$ oc patch network.operator cluster --patch "$(cat network.yaml)" --type=merge
```

Comment:
Unfortunately, if you have existing additionalNetworks, this command is going to overwrite them with the one you specified in `network.yaml` instead of merging them as expected.

Itzik opened https://issues.redhat.com/browse/OCPBUGS-4515. We may to want to either provide a different command that maintains existing additional networks or document the limitation. @dulek, WDYT?

Reply:
Oh, I solved something similar recently with a command like this:

```
        oc get -n $ns $resource $name -o json | jq -Mcr "if .metadata.finalizers != null then del(.metadata.finalizers[] | select(. == \"${finalizer}\")) else . end" | oc replace -n $ns $resource $name -f -
```

What this does is - get the resource from API, manipulate it's representation using `jq` (in this case remove a particular finalizer) and then use `oc replace` to make sure we don't suffer from lost-update issue.

I wonder why we won't just recommend using `oc edit` instead to add the network?

Reply:
@dulek I can add whatever you and QE propose and approve with few exceptions. 

One note: We really don't like using `jq` in commands. If you can swap `jq` for `jsonpath` in there, I think I could include it.

Reply:
I don't think `jsonpath` would do here, we don't want to get a particular object from JSON K8s object, but rather to modify it.

Reply:
@dulek I'll argue for that if pushed about it, then. 👍

Reply:
@mandre @itzikb-redhat +1/-1 on @dulek's suggestion? 

I'm also curious about `oc edit` as an option here, too.

Reply:
Yes, `oc edit` to add the needed entry in additionalNetworks is safer.

Reply:
Makes total sense to me. And leave finding the non-interactive command as an exercise for the reader.